ddtrace/contrib/datadog_lambda/patch.py

```diff
@@ -0,0 +1,89 @@
+from importlib import import_module
+import os
+import signal
+
+from ddtrace.internal.wrapping import unwrap
+from ddtrace.internal.wrapping import wrap
+
+
+class ImpendingTimeout(Exception):
+    pass
+
+
+def crash_flush(_, __):
+    """Raises an `ImpendingTimeout` error which causes the tracer to shutdown."""
+    # TODO: Update to tag error in local root parent instead of raising the error.
+
+    raise ImpendingTimeout("Datadog detected an Impending Timeout")
+
+
+def check_timeout(context):
+    """
+    Creates a timeout to detect when an AWS Lambda handler's remaining
+    time is about to end.
+
+    Crashes flushes when the signal is activated
+    """
+    signal.signal(signal.SIGALRM, crash_flush)
+    remaining_time_in_millis = context.get_remaining_time_in_millis()
+    apm_flush_deadline = int(os.environ.get("DD_APM_FLUSH_DEADLINE", 0))
```

Comment:
Is the DD_APM_FLUSH_DEADLINE configuration specific to tracers? If so we should rename it to `DD_TRACE_LAMBDA_FLUSH_DEADLINE`. We should also use a consistent configuration across tracers.

Reply:
It's not specific to tracers but to timeouts in lambdas, it has already been set like that for `dd-trace-js`, so in order to keep consistency, we have to keep the env var as is.

Reply:
Hey, yes. I just removed my previous comment, since every APM tracer should support deadlines. 
>Containers, k8s, and regular ec2 instances have kill notices and context deadlines

In `dd-trace-js`, `DD_APM_FLUSH_DEADLINE` is already set like that